### PR TITLE
Fix the issue for NamedTemporaryFile() on Windows

### PR DESCRIPTION
**Pull request by hefengxian**
Since last change about `NamedTemporaryFile()` function's option `delete=False` the temporary file will keep in tmp directory when program was finished. this PR for delete those temporary files automatically

### CODE REVIEW

File: farasa/__base.py [hunk 1]
```diff
@@ -112,18 +113,28 @@ def _check_java_version(self):
...
-                print("error occurred!", proc.stderr)
+                print("error occurred! stdout: ", proc.stdout, ' stderr: ', proc.stderr)
                 print("return code:", proc.returncode)
-                raise Exception('Internal Error occured!')
+                raise Exception('Internal Error occurred!')
+        finally:
+            itmp.close()
+            otmp.close()
+            os.unlink(itmp.name)
+            os.unlink(otmp.name)
```
Comment (MagedSaeed) [inline farasa/__base.py:136]: Is this part platform independent? That is interesting!

Comment (hefengxian) [reply farasa/__base.py:136]: I checked Python docs, `os.unlink()` is the same with `os.remove()` and it Unix & Windows both available

But there is an option to use `pathlib.Path.unlink(file.name)` check here 

[https://stackoverflow.com/questions/42636018/python-difference-between-os-remove-and-os-unlink-and-which-one-to-use](https://stackoverflow.com/questions/42636018/python-difference-between-os-remove-and-os-unlink-and-which-one-to-use)

**Comment by hefengxian** [comment]
> I think this still will have the issue that the temporary file will not be deleted on program crash or CTRL+C signal, does not it?
> 
> How about applying some ideas from [here](https://stackoverflow.com/questions/3850261/doing-something-before-program-exit)[https://stackoverflow.com/questions/3850261/doing-something-before-program-exit]?
> 
> How about this snippet?
> 
> ```
> try:
>     ...
> except KeyboardInterrupt:
>     # clean up
>     exit()
> ```

I have on idea about this, but i think this is not a good solution, sometimes i'm using PyCharm run & stop button, i'm not sure what signal is send by stop button; but also cause the temporary file not removed

**Comment by hefengxian** [comment]
I have test in those platforms

- macOS
    + OS version: `catalina`
    + Java version: `11.0`
    + Result: `pass`
- Ubuntu Server
    + OS version: `18.04.2`
    + Java version: `1.8`
    + Result: pass
- Windows
    + OS version: `10`
    + Java version: `1.8`
    + Result: `pass`

and tmp dir no file left

**Comment by MagedSaeed** [comment]
I think we should merge, then update the pypi package with what we have right now. Further modifications will be  added later. What do you think?

**Comment by MagedSaeed** [comment]
I am merging the pull request right now.

**Pull request merged**